src/logic/safe/transactions/pendingTxMonitor.ts

```diff
@@ -50,20 +52,32 @@ export const pendingTxsMonitor = async (): Promise<void> => {
 
   const web3 = getWeb3()
 
+  let sessionBlockNumber: number
   try {
-    const sessionBlockNumber = await web3.eth.getBlockNumber()
+    sessionBlockNumber = await web3.eth.getBlockNumber()
+  } catch {
+    // Don't continue if there was an error retrieving current block number
+    return
+  }
 
-    // Exponentially watch each pending transaction for (un-)successful mine within 50 blocks
-    await Promise.all(
-      pendingTxs.map(([txId, txHash]) => {
-        backOff(() => isPendingTxMined(sessionBlockNumber, txId, txHash), {
-          startingDelay: START_DELAY,
+  for (const [txId, txHash] of pendingTxs) {
+    let currentAttempt = 0
+
+    try {
+      await backOff(
+        () => {
+          currentAttempt++
+          return isPendingTxMined(sessionBlockNumber, txId, txHash, hasReachedMaxAttempts(currentAttempt))
+        },
+        {
+          startingDelay: INITIAL_TIMEOUT,
+          timeMultiple: TIMEOUT_MULTIPLIER,
           numOfAttempts: MAX_ATTEMPTS,
-          retry: () => !hasReachedMaxAttempts(),
-        })
-      }),
-    )
-  } catch {
-    // Ignore
+          retry: () => !hasReachedMaxAttempts(currentAttempt),
```

Comment:
[Retry takes the attempt counter as an argument](https://github.com/coveooss/exponential-backoff#ibackoffoptions), so you don't need a separate counter.